### PR TITLE
standardize checking to see if service is supported Fixed #1453

### LINKED ISSUE #1442: standardize required environment variables for services
Services should be able to specify `REQUIRED_ENV` values which are then checked and handled uniformly by the rpc manager

### LINKED ISSUE #1453: standardize checking service support
Services should be able to specify custom environment checking functions to see if they can be loaded.

### PR DESCRIPTION
**Pull request by hamidzr**

### CODE REVIEW

File: src/server/rpc/rpc-manager.js [hunk 1]
```diff
@@ -39,8 +39,14 @@ RPCManager.prototype.loadRPCs = function() {
         .map(name => [name, path.join(PROCEDURES_DIR, name, name+'.js')])
         .filter(pair => fs.existsSync(pair[1]))
         .map(pair => [pair[0], require(pair[1])])
-        .filter(pair => typeof pair[1] === 'function' ||
-            (!!pair[1] && !_.isEmpty(pair[1])))
+        .filter(pair => {
+            let service = pair[1];
+            if (typeof service === 'function' || !!service && !_.isEmpty(service)) {
+                if(typeof service.isSupported === 'function') return service.isSupported();
```
Comment (brollb) [inline src/server/rpc/rpc-manager.js]: this could be simplified by just checking `if (service.isSupported) return service.isSupported()`. We shouldn't have to check if it is a function because our dev API for creating RPCs dictates that it has to be a function. That is, if it isn't a function, then it is a bug and silently failing might actually be a bad thing - having the server crash would give the developer faster feedback about mistakes in the RPC.

**Comment by brollb** [comment]
Also, this isn't exactly what #1442 was referring to (although it is very closely related). Basically, #1442 was trying to make a convenient way of specifying that an environment variable was needed so the rpc-manager could detect and not load the service (and print a console message) whereas this PR adds support for custom support functions.

In other words, this is a good and more generic solution but it would be nice to have a standard approach for just checking for api keys (rather than having a bunch of duplicated code that just checks for a slightly different API key). Now, if all our services which require an api key use the api consumer, then we might not need anything more than this PR (but the other services should be updated to use the same method and then we can consider that the convenience approach - this may be the best approach).

tl;dr - I might move this to a different issue since the focus on #1442 was actually trying to standardize the duplicated environment variable checking done in many of the API services rather than adding support for custom support functions for services.

File: src/server/rpc/rpc-manager.js (hunk 1 above)
Comment (brollb) [reply src/server/rpc/rpc-manager.js]: Also, could you print a console message saying that the given service is not going to be loaded?

**Comment by brollb** [comment]
#1453

File: src/server/rpc/rpc-manager.js (hunk 1 above)
Comment (hamidzr) [reply src/server/rpc/rpc-manager.js]: the check to see if service is a fn is already in there and I'm trying to make as little a change as possible in this PR
